keras_cv/layers/preprocessing/random_translation.py

```diff
@@ -0,0 +1,237 @@
+
+    def augment_image(self, image, transformation, **kwargs):
+        return self._translate_image(image, transformation, self.interpolation)
+
+    def get_random_transformation(self, **kwargs):
+        translate_horizontal = 0.0
+        translate_vertical = 0.0
+        if self.x_factor:
+            translate_horizontal = self.x_factor()
+        if self.y_factor:
+            translate_vertical = self.y_factor()
+        return {
+            "translate_horizontal": tf.cast(translate_horizontal, dtype=tf.float32),
+            "translate_vertical": tf.cast(translate_vertical, dtype=tf.float32),
+        }
+
+    def _translate_image(self, image, transformation, interpolation_method):
+        image = preprocessing.ensure_tensor(image, self.compute_dtype)
+        image_shape = image.shape
+        image_width = tf.cast(image_shape[1], tf.float32)
+        image_height = tf.cast(image_shape[0], tf.float32)
+        element = tf.stack(
+            [
+                transformation["translate_horizontal"] * image_width,
+                transformation["translate_vertical"] * image_height,
+            ],
+        )
+        output = preprocessing.transform(
+            tf.expand_dims(image, 0),
+            preprocessing.get_translation_matrix(tf.expand_dims(element, axis=0)),
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            interpolation=interpolation_method,
+        )
+        output = tf.squeeze(output, 0)
+        output.set_shape(image_shape)
+        return output
+
+    @staticmethod
+    def _translate_bounding_boxes(bounding_boxes, dx, dy):
+        x1, x2, x3, x4, rest = tf.split(
+            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
+        )
+        output = tf.stack(
+            [
+                x1 + dx,
+                x2 + dy,
+                x3 + dx,
+                x4 + dy,
+                rest,
+            ],
+            axis=-1,
+        )
+        output = tf.squeeze(output, axis=1)
+        return output
+
+    def augment_bounding_boxes(
+        self, bounding_boxes, transformation=None, image=None, **kwargs
+    ):
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomTranslation()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomTranslation(bounding_box_format='xyxy')`"
+            )
+
+        bounding_boxes = bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="rel_xyxy",
+            images=image,
+        )
+        bounding_boxes = RandomTranslation._translate_bounding_boxes(
+            bounding_boxes,
+            transformation["translate_horizontal"],
+            transformation["translate_vertical"],
+        )
+
```

Comment:
Can we add a unit test covering the case where a bounding box is translated fully out of the image?